pkg/container/cri/types.go

```diff
@@ -49,6 +49,20 @@ type Mount struct {
 	Propagation MountPropagation `protobuf:"varint,5,opt,name=propagation,proto3,enum=runtime.v1alpha2.MountPropagation" json:"propagation,omitempty"`
 }
 
+// PortMapping specifies a host port mapped into a container port.
+// In yaml this looks like:
+//  containerPort: 80
+//  hostPort: 8000
+//  listenAddress: 127.0.0.1
+type PortMapping struct {
+	// Port within the container.
+	ContainerPort int32 `protobuf:"varint,1,opt,name=container_port,json=containerPort,proto3" json:"containerPort,omitempty"`
+	// Port on the host.
+	HostPort int32 `protobuf:"varint,2,opt,name=host_path,json=hostPort,proto3" json:"hostPort,omitempty"`
+	// TODO: add protocol (tcp/udp) and port-ranges
+	ListenAddress string `protobuf:"bytes,3,opt,name=listenAddress,json=hostPort,proto3" json:"listenAddress,omitempty"`
```

Comment:
I don't know if this should be the same address that we are using for the apiserveraddress
https://github.com/kubernetes-sigs/kind/blob/edbae5fdd1f1670eef0323018ccb18ced552fd29/pkg/cluster/config/types.go#L96

It seems unlikely that someone wants to use different addresses for forwarding ports, thoughts?

Reply:
The current behavior is that it will listen on `0.0.0.0` (since this is the behavior of `docker run --publish` if no listenAddress is given). :thinking: Maybe we can default this value to `APIServerAddress` if no ListenAddress is specified, but not sure what is best.

Reply:
@BenTheElder can you help here? ^^^

Reply:
I think it's reasonable to match docker here for now, with this field you've specifically asked to forward a port, whereas the apiServer port kind is doing even if you don't know about that as a user. Since the user may not even know, we're a bit warier about exposing it beyond localhost by default (also that port is definitely hosting something powerful that should be secured).

In a future API revision we can change the defaulting easily since we're still alpha :sweat_smile: